### PR TITLE
Add sink config options

### PR DESCRIPTION
**Pull request by raminqaf**
@1p4pk and @Shark I refactored the sink code and added the config class for the sink as well. It is now possible to set the write buffer size and toggle the checkpoint data point write. I also passed the `InfluxDBConfig` class properties (URL, Bucket, Org, Username, Password,...) to the SinkOptions as well and have a central config class for all of that.
Fixes https://www.notion.so/Add-Configuration-Parameters-for-Sink-219d7b55b8b0435983161254790e6158

### CODE REVIEW

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSinkOptions.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import org.apache.flink.streaming.connectors.influxdb.common.InfluxDBOptionsBase;
+
+public final class InfluxDBSinkOptions extends InfluxDBOptionsBase {
+
+    private InfluxDBSinkOptions() {}
+
+    public static final ConfigOption<Boolean> WRITE_DATA_POINT_CHECKPOINT =
+            ConfigOptions.key("write.data_point.checkpoint")
+                    .booleanType()
+                    .defaultValue(true)
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSinkOptions.java]: I think default should be false

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java [hunk 2]
```diff
@@ -124,14 +109,20 @@
      * @return a InfluxDBSource with the settings made for this builder.
      */
     public InfluxDBSource<OUT> build() {
-        this.sanityCheck();
         return new InfluxDBSource<>(this.properties, this.deserializationSchema);
     }
 
     // ------------- private helpers  --------------
-    private void sanityCheck() {
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java:132]: Why did you delete the sanityCheck here?

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSinkBuilder.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import static org.apache.flink.streaming.connectors.influxdb.sink.InfluxDBSinkOptions.INFLUXDB_URL;
+import static org.apache.flink.streaming.connectors.influxdb.sink.InfluxDBSinkOptions.INFLUXDB_USERNAME;
+import static org.apache.flink.streaming.connectors.influxdb.sink.InfluxDBSinkOptions.WRITE_BUFFER_SIZE;
+import static org.apache.flink.streaming.connectors.influxdb.sink.InfluxDBSinkOptions.WRITE_DATA_POINT_CHECKPOINT;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+import java.util.Properties;
+import org.apache.flink.streaming.connectors.influxdb.sink.writer.InfluxDBSchemaSerializer;
+
+public class InfluxDBSinkBuilder<IN> {
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/InfluxDBSinkBuilder.java]: final? ;)

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java [hunk 4]
```diff
@@ -38,9 +41,11 @@
 public class InfluxDBCommitter implements Committer<Long> {
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java]: final? ;)

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java [hunk 5]
```diff
@@ -42,24 +46,26 @@
 @Slf4j
 public class InfluxDBWriter<IN> implements SinkWriter<IN, Long, Point> {
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java]: final? ;)

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java [hunk 6]
```diff
@@ -17,8 +17,6 @@
  */
 package org.apache.flink.streaming.connectors.influxdb.source;
 
-import static org.apache.flink.util.Preconditions.checkNotNull;
-
 import java.util.Properties;
 import org.apache.flink.streaming.connectors.influxdb.source.reader.deserializer.InfluxDBDataPointDeserializer;
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java:24]: final?

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java (hunk 2 above)
Comment (raminqaf) [reply flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/InfluxDBSourceBuilder.java:132]: Ups! I was testing something and removed it. I will add it back.

File: flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java [hunk 7]
```diff
@@ -61,12 +66,14 @@ public InfluxDBCommitter(final InfluxDBConfig config) {
     @SneakyThrows
     @Override
     public List<Long> commit(final List<Long> committables) {
-        log.info("A checkpoint is set.");
-        Optional<Long> lastTimestamp = Optional.empty();
-        if (committables.size() >= 1) {
-            lastTimestamp = Optional.ofNullable(committables.get(committables.size() - 1));
+        if (this.writeCheckpoint) {
+            log.info("A checkpoint is set.");
```
Comment (1p4pk) [inline flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/commiter/InfluxDBCommitter.java]: I think better to use log.trace or log.debug

**Review by 1p4pk — APPROVED** [review]
LGTM
